utils/common/src/main/kotlin/FileMatcher.kt

```diff
@@ -39,13 +40,42 @@ class FileMatcher(
 ) {
     constructor(vararg patterns: String, ignoreCase: Boolean = false) : this(patterns.asList(), ignoreCase)
 
-    private val matcher = AntPathMatcher().apply {
-        setCaseSensitive(!ignoreCase)
+    companion object {
```

Comment:
For the fun of it, here's an alternative:

```
package org.ossreviewtoolkit.utils.common

import org.springframework.util.AntPathMatcher

private val matcherMap = mapOf(
    false to AntPathMatcher().apply { setCaseSensitive(true) }::match,
    true to AntPathMatcher().apply { setCaseSensitive(false) }::match
)

/**
 * A class to determine whether a path is matched by any of the given globs.
 */
class FileMatcher(
    /**
     * The collection of [glob patterns][1] to consider for matching.
     *
     * [1]: https://docs.oracle.com/javase/tutorial/essential/io/fileOps.html#glob
     */
    val patterns: Collection<String>,

    /**
     * Toggle the case-sensitivity of the matching.
     */
    ignoreCase: Boolean = false
) {
    constructor(vararg patterns: String, ignoreCase: Boolean = false) : this(patterns.asList(), ignoreCase)

    private val matcher = matcherMap.getValue(ignoreCase)

    /**
     * Return true if [path] is matched by any of [patterns], false otherwise. The [path] must use '/' as separators,
     * if it contains any.
     */
    fun matches(path: String): Boolean = patterns.any { pattern -> matcher(pattern, path) }
}
```

Reply:
I made use of the function references but I prefer two properties for the matchers over a map.